download-basket/src/main/java/downloadbasket/helpers/SendDownloadDetailsToEmailThread.java

```diff
@@ -0,0 +1,289 @@
+package downloadbasket.helpers;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.StringWriter;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.Hashtable;
+import java.util.Properties;
+import java.util.UUID;
+import java.util.Locale;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+import java.util.zip.ZipOutputStream;
+
+import javax.mail.Address;
+import javax.mail.Message;
+import javax.mail.Transport;
+import javax.mail.internet.MimeBodyPart;
+import javax.mail.internet.MimeMessage;
+import javax.xml.bind.JAXBException;
+
+import fi.nls.oskari.domain.map.OskariLayer;
+import fi.nls.oskari.log.LogFactory;
+import fi.nls.oskari.log.Logger;
+import fi.nls.oskari.map.layer.OskariLayerService;
+import fi.nls.oskari.map.layer.OskariLayerServiceIbatisImpl;
+import fi.nls.oskari.util.PropertyUtil;
+import downloadbasket.data.LoadZipDetails;
+import downloadbasket.data.NormalWayDownloads;
+import downloadbasket.data.ZipDownloadDetails;
+import org.apache.commons.mail.HtmlEmail;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.sun.mail.smtp.SMTPTransport;
+
+/**
+ * Send download details email service (thread).
+ */
+public class SendDownloadDetailsToEmailThread implements Runnable {
+	JSONArray downLoadDetails;
+	JSONObject userDetails;
+	Locale locale;
+
+	private final Logger LOGGER = LogFactory.getLogger(SendDownloadDetailsToEmailThread.class);
+	private final String PARAM_CROPPING_MODE = "croppingMode";
+	private final String PARAM_CROPPING_LAYER = "croppingLayer";
+	private final String PARAM_LAYER = "layer";
+	private final String PARAM_LAYER_ID = "id";
+
+	/**
+	 * Constructor.
+	 * 
+	 * @param downLoadDetails
+	 *            download details
+	 * @param userDetails
+	 *            user details
+	 */
+	public SendDownloadDetailsToEmailThread(JSONArray downLoadDetails, JSONObject userDetails, Locale locale) {
+		this.downLoadDetails = downLoadDetails;
+		this.userDetails = userDetails;
+		this.locale = locale;
+	}
+
+	/**
+	 * 
+	 * Overrides the run method. Collects the download materials and sends them
+	 * using the variables given in constructor.
+	 * 
+	 */
+	@Override
+	public void run() {
+
+		try {
+			DownloadServices ds = new DownloadServices();
+			ArrayList<ZipDownloadDetails> mergeThese = new ArrayList<ZipDownloadDetails>();
+			final String strTempDir = PropertyUtil.get("oskari.wfs.download.folder.name");
+			String normalWayDownload = PropertyUtil.get("oskari.wfs.download.normal.way.downloads");
+			String[] temp = normalWayDownload.split(",");
+			NormalWayDownloads normalDownloads = new NormalWayDownloads();
+			for (int i = 0; i < temp.length; i++) {
+				normalDownloads.addDownload(temp[i]);
+			}
+
```

Comment:
What's the difference between these two properties? Both are used to populate "NormalWayDownloads":

 		String[] bboxDownloadTypes = PropertyUtil.getCommaSeparatedList("oskari.wfs.bbox.downloads");
 		NormalWayDownloads normalDownloads = new NormalWayDownloads();
 		for (String download : bboxDownloadTypes) {
 			normalDownloads.addDownload(download);
 		}
 		...
		String normalWayDownload = PropertyUtil.get("oskari.wfs.download.normal.way.downloads");
		String[] temp = normalWayDownload.split(",");
		NormalWayDownloads normalDownloads = new NormalWayDownloads();
		for (int i = 0; i < temp.length; i++) {
			normalDownloads.addDownload(temp[i]);
		}

Can you update the PR description to include any properties and configurations that this functionality uses.

Reply:
They both use oskari.wfs.download.normal.way.downloads, fixed.